pkg/pulumiyaml/ast/template.go

```diff
@@ -698,6 +699,127 @@ func (d *TemplateDecl) NewDiagnosticWriter(w io.Writer, width uint, color bool)
 	return newDiagnosticWriter(w, fileMap, width, color)
 }
 
+func parseTypeSpec(configDecl *ConfigParamDecl) (schema.TypeSpec, error) {
+	typeSpec := schema.TypeSpec{}
+	if configDecl.Type == nil {
+		return typeSpec, fmt.Errorf("missing type")
+	}
+	switch configDecl.Type.Value {
+	case "string":
+		typeSpec.Type = "string"
+	case "integer":
+		typeSpec.Type = "integer"
+	case "boolean":
+		typeSpec.Type = "boolean"
+	case "array":
+		if configDecl.Items == nil {
+			return typeSpec, fmt.Errorf("missing items")
+		}
+		itemsTypeSpec, err := parseTypeSpec(configDecl.Items)
+		if err != nil {
+			return typeSpec, err
+		}
+		typeSpec.Type = "array"
+		typeSpec.Items = &itemsTypeSpec
+	default:
+		return typeSpec, fmt.Errorf("unknown type: %s", configDecl.Type.Value)
+	}
+	return typeSpec, nil
+}
+
+func (d *TemplateDecl) GenerateSchema() (schema.PackageSpec, error) {
+	description := ""
+	if d.Description != nil {
+		description = d.Description.Value
+	}
+	schemaDef := schema.PackageSpec{
+		Name:        d.Name.Value,
+		Description: description,
+		Language: map[string]schema.RawMessage{
+			"nodejs": schema.RawMessage(`{"respectSchemaVersion": true}`),
+			"python": schema.RawMessage(`{"respectSchemaVersion": true}`),
+			"cshap":  schema.RawMessage(`{"respectSchemaVersion": true}`),
+			"java":   schema.RawMessage(`{"respectSchemaVersion": true}`),
+			"go":     schema.RawMessage(`{"respectSchemaVersion": true}`),
+		},
+	}
+
+	resourcesDef := make(map[string]schema.ResourceSpec)
+	for _, component := range d.Components.Entries {
```

Comment:
Did I miss a PR where we introduced d.Components?

Reply:
It was introduced here: https://github.com/pulumi/pulumi-yaml/pull/747 Looks like I messed up tagging you there, sorry about that.